<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>For HKCRC Crane Testing</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="messagesGroup">
      <property name="title">
       <string>Messages</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QPlainTextEdit" name="messageTextEdit">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="bottomWidget" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QGroupBox" name="parametersGroup">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>3</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Modbus Parameters</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Parameter</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Address</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Value</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="controlModeLabel">
            <property name="text">
             <string>Control Mode</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="controlModeAddrSpinBox">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="displayIntegerBase">
             <number>16</number>
            </property>
            <property name="prefix">
             <string>0x</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QSpinBox" name="controlModeSpinBox">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="slewSpeedLabel">
            <property name="text">
             <string>Slew Speed</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="slewSpeedAddrSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="displayIntegerBase">
             <number>16</number>
            </property>
            <property name="prefix">
             <string>0x</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QSpinBox" name="slewSpeedSpinBox">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="luffSpeedLabel">
            <property name="text">
             <string>Luff Speed</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="luffSpeedAddrSpinBox">
            <property name="minimum">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="displayIntegerBase">
             <number>16</number>
            </property>
            <property name="prefix">
             <string>0x</string>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QSpinBox" name="luffSpeedSpinBox">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="hoistSpeedLabel">
            <property name="text">
             <string>Hoist Speed</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QSpinBox" name="hoistSpeedAddrSpinBox">
            <property name="minimum">
             <number>3</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="displayIntegerBase">
             <number>16</number>
            </property>
            <property name="prefix">
             <string>0x</string>
            </property>
           </widget>
          </item>
          <item row="4" column="2">
           <widget class="QSpinBox" name="hoistSpeedSpinBox">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="pathLengthLabel">
            <property name="text">
             <string>Path Length</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QSpinBox" name="pathLengthAddrSpinBox">
            <property name="minimum">
             <number>4</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="displayIntegerBase">
             <number>16</number>
            </property>
            <property name="prefix">
             <string>0x</string>
            </property>
           </widget>
          </item>
          <item row="5" column="2">
           <widget class="QSpinBox" name="pathLengthSpinBox">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="slewPathLabel">
            <property name="text">
             <string>Slew Path</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QSpinBox" name="slewPathAddrSpinBox">
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="displayIntegerBase">
             <number>16</number>
            </property>
            <property name="prefix">
             <string>0x</string>
            </property>
           </widget>
          </item>
          <item row="6" column="2">
           <widget class="QSpinBox" name="slewPathSpinBox">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="luffPathLabel">
            <property name="text">
             <string>Luff Path</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QSpinBox" name="luffPathAddrSpinBox">
            <property name="minimum">
             <number>6</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="displayIntegerBase">
             <number>16</number>
            </property>
            <property name="prefix">
             <string>0x</string>
            </property>
           </widget>
          </item>
          <item row="7" column="2">
           <widget class="QSpinBox" name="luffPathSpinBox">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="hoistPathLabel">
            <property name="text">
             <string>Hoist Path</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QSpinBox" name="hoistPathAddrSpinBox">
            <property name="minimum">
             <number>7</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="displayIntegerBase">
             <number>16</number>
            </property>
            <property name="prefix">
             <string>0x</string>
            </property>
           </widget>
          </item>
          <item row="8" column="2">
           <widget class="QSpinBox" name="hoistPathSpinBox">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
           </widget>
          </item>
          <item row="9" column="1" colspan="2">
           <widget class="QPushButton" name="sendButton">
            <property name="text">
             <string>Send</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="connectionGroup">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumWidth">
          <number>300</number>
         </property>
         <property name="maximumWidth">
          <number>300</number>
         </property>
         <property name="title">
          <string>Connection Settings</string>
         </property>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="ipLabel">
            <property name="text">
             <string>Server IP:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="ipEdit">
            <property name="text">
             <string>192.168.1.100</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="portLabel">
            <property name="text">
             <string>Port:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="portSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="value">
             <number>502</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="slaveLabel">
            <property name="text">
             <string>Slave Address:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="slaveSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>247</number>
            </property>
            <property name="value">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="registerTypeLabel">
            <property name="text">
             <string>Register Type:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QComboBox" name="registerTypeCombo">
            <item>
             <property name="text">
              <string>Holding Registers</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Input Registers</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Coils</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Discrete Inputs</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QPushButton" name="connectButton">
            <property name="text">
             <string>Connect</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>